### PR TITLE
Added mock-nifi, cert details and integration test infra

### PR DESCRIPTION
**Pull request by steveburton4**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -56,6 +56,43 @@ jobs:
...
+            sudo apt-get install openjdk-8-jdk
+
+            export HBASE_TO_MONGO_EXPORT_VERSION=$(cat ./gradle.properties | cut -f2 -d'=')
+            export DATA_KEY_SERVICE_URL=http://localhost:8090
+            export DATA_KEY_SERVICE_URL_SSL=https://localhost:8091
+            export S3_BUCKET=dummy-bucket
+            export S3_PREFIX_FOLDER=dummy-folder
+            export AWS_DEFAULT_REGION=dummy-region
+            export AWS_ACCESS_KEY_ID=dummy-key
+            export AWS_SECRET_ACCESS_KEY=dummy-secret
```
Comment (steveburton4) [inline .circleci/config.yml:76]: These might not be relevant for snapshot sender. Might need changing according to what is needed.

File: Makefile [hunk 2]
```diff
@@ -38,26 +40,49 @@ build-base-images: ## Build base images to avoid rebuilding frequently
...
+	@{ \
+		export HBASE_TO_MONGO_EXPORT_VERSION=$(hbase_to_mongo_version); \
+		export AWS_DEFAULT_REGION=$(aws_default_region); \
+		export AWS_ACCESS_KEY_ID=$(aws_access_key_id); \
+		export AWS_SECRET_ACCESS_KEY=$(aws_secret_access_key); \
+		export S3_BUCKET=$(s3_bucket); \
+		export S3_PREFIX_FOLDER=$(s3_prefix_folder); \
+		export DATA_KEY_SERVICE_URL=$(data_key_service_url); \
+		export DATA_KEY_SERVICE_URL_SSL=$(data_key_service_url_ssl); \
+		echo "Waiting for exporters"; \
```
Comment (steveburton4) [inline Makefile]: Again env variables likely need changing

File: resources/Dockerfile_mock_nifi [hunk 3]
```diff
@@ -12,4 +12,7 @@ RUN pip3 install -r requirements.txt
 
 COPY mock-nifi/* ./
 
+COPY mock-nifi-keystore.jks /ssl/keystore.jks
+COPY mock-nifi-truststore.crt /ssl/truststore.jks
+
 CMD ["python3", "mocknifi.py"]
```
Comment (steveburton4) [inline resources/Dockerfile_mock_nifi]: Needs to download the jar for new mock nifi repo and run it -> use the htme dockerfile as an example

File: src/integration/groovy/IntegrationTest.groovy [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            println(object.getClass())
+            def timestamp = object.get('timestamp')
+            lineCount++
+            log.info("lineCount = " + lineCount)
+            log.info("timestamp = " + timestamp)
+            log.info("Checking timestamp = " + expectedTimestamp)
+            assert timestamp == expectedTimestamp
+        }
+        log.info("Checking lineCount = " + expectedLineCount)
+        lineCount == expectedLineCount
```
Comment (steveburton4) [inline src/integration/groovy/IntegrationTest.groovy]: This is the biggest change. This currently checks hbase and ensures the data has been written. Instead needs to check mock nifi that the files are in the output folder as requested.

**Review by benjaminsherwood — CHANGES_REQUESTED** [review]
Blocking

File: .circleci/config.yml (hunk 1 above)
Comment (benjaminsherwood) [reply .circleci/config.yml:76]: removed HBASE_TO_MONGO_EXPORT_VERSION

File: src/integration/groovy/IntegrationTest.groovy (hunk 4 above)
Comment (DanielChicot) [reply src/integration/groovy/IntegrationTest.groovy]: Done in integration_test.py.

**Review by benjaminsherwood — APPROVED** [review]

**Pull request merged**